### PR TITLE
test(functional): rewrite oauth permissions tests

### PR DESCRIPTION
**Pull request by ashrivastava-qa**
## Because

As part of the playwright test migration, the [oauth permissions](https://github.com/pdehaan/fxa/blob/0ae60a1df256332e8581898db39b0e4997bab7b4/packages/fxa-content-server/tests/functional/oauth_permissions.js) and [oauth prompt none](https://github.com/pdehaan/fxa/blob/0ae60a1df256332e8581898db39b0e4997bab7b4/packages/fxa-content-server/tests/functional/oauth_prompt_none.js) have been rewritten in this PR.

## This pull request

contains the tests identified as 'Keep' in the [analysis sheet](https://docs.google.com/spreadsheets/d/11Wq-Y-ipeNFXqLHbr3GJCh_f_qEAG-CUuqBt5Dcnh5k/edit#gid=1966588528) for [oauth permissions](https://github.com/pdehaan/fxa/blob/0ae60a1df256332e8581898db39b0e4997bab7b4/packages/fxa-content-server/tests/functional/oauth_permissions.js) and [oauth prompt none](https://github.com/pdehaan/fxa/blob/0ae60a1df256332e8581898db39b0e4997bab7b4/packages/fxa-content-server/tests/functional/oauth_prompt_none.js)

## Issue that this pull request solves

Closes: FXA-5917 FXA-5920

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

Any other information that is important to this pull request.

### CODE REVIEW

File: packages/functional-tests/tests/oauth/oauthPermissions.spec.ts [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+      } catch (e) {
+        // Handle the error here
+        console.error('An error occurred during account cleanup:', e);
+        // Optionally, rethrow the error to propagate it further
+        throw e;
+      }
+    }
+  });
+
+  test('signup without `prompt=consent', async ({
```
Comment (pdehaan) [inline packages/functional-tests/tests/oauth/oauthPermissions.spec.ts:31]: nit, the quotes seem mismatched.
```suggestion
  test('signup without `prompt=consent`', async ({
```

File: packages/functional-tests/tests/oauth/oauthPermissions.spec.ts [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+    //Verify permissions header
+    expect(await login.permissionsHeader()).toBe(true);
+    await login.acceptOauthPermissions();
+
+    //Verify sign up code header
+    expect(await login.isSignUpCodeHeader()).toBe(true);
+  });
+
+  test('signin without `prompt=consent', async ({
```
Comment (pdehaan) [inline packages/functional-tests/tests/oauth/oauthPermissions.spec.ts:63]:
```suggestion
  test('signin without `prompt=consent`', async ({
```

File: packages/functional-tests/tests/oauth/oauthPermissions.spec.ts [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    });
+    await relier.goto();
+    await relier.clickEmailFirst();
+    await login.fillOutEmailFirstSignIn(email, password);
+
+    //Verify logged in to relier
+    expect(await relier.isLoggedIn()).toBe(true);
+  });
+
+  test('signin with `prompt=consent', async ({
```
Comment (pdehaan) [inline packages/functional-tests/tests/oauth/oauthPermissions.spec.ts:79]:
```suggestion
  test('signin with `prompt=consent`', async ({
```

File: packages/functional-tests/tests/oauth/oauthPermissions.spec.ts [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    expect(await login.isSignUpCodeHeader()).toBe(true);
+  });
+
+  test('signin without `prompt=consent', async ({
+    target,
+    pages: { login, relier },
+  }) => {
+    await target.auth.signUp(email, password, {
+      lang: 'en',
+      preVerified: 'true',
```
Comment (pdehaan) [inline packages/functional-tests/tests/oauth/oauthPermissions.spec.ts:69]: Q: Should this be a string  `'true'` or Boolean `true`?

File: packages/functional-tests/tests/oauth/oauthPermissions.spec.ts [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+    });
+    const query = new URLSearchParams({
+      prompt: 'consent',
+      email: email,
+    });
+    await page.goto(target.relierUrl + `/?${query.toString()}`);
```
Comment (pdehaan) [inline packages/functional-tests/tests/oauth/oauthPermissions.spec.ts:166]:
```suggestion
    await page.goto(`${target.relierUrl}/?${query.toString()}`);
```

File: packages/functional-tests/tests/oauth/oauthPromptNone.spec.ts [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+  test('fails if no user logged in', async ({
+    page,
+    target,
+    pages: { relier },
+  }) => {
+    const query = new URLSearchParams({
+      login_hint: email,
+      return_on_error: 'false',
+    });
+    await page.goto(target.relierUrl + `/?${query.toString()}`);
```
Comment (pdehaan) [inline packages/functional-tests/tests/oauth/oauthPromptNone.spec.ts:26]:
```suggestion
    await page.goto(`${target.relierUrl}/?${query.toString()}`);
```

File: packages/functional-tests/tests/oauth/oauthPromptNone.spec.ts [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+    const query = new URLSearchParams({
+      client_id: '7f368c6886429f19', // eslint-disable-line camelcase
+      forceUA:
+        'Mozilla/5.0 (Linux; Android 6.0; Nexus 5 Build/MRA58N) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/65.0.3325.181 Mobile Safari/537.36',
+      // eslint-disable-next-line camelcase
+      keys_jwk:
+        'eyJrdHkiOiJFQyIsImtpZCI6Im9DNGFudFBBSFZRX1pmQ09RRUYycTRaQlZYblVNZ2xISGpVRzdtSjZHOEEiLCJjcnYiOi' +
+        'JQLTI1NiIsIngiOiJDeUpUSjVwbUNZb2lQQnVWOTk1UjNvNTFLZVBMaEg1Y3JaQlkwbXNxTDk0IiwieSI6IkJCWDhfcFVZeHpTaldsdX' +
+        'U5MFdPTVZwamIzTlpVRDAyN0xwcC04RW9vckEifQ',
+      login_hint: email,
```
Comment (pdehaan) [inline packages/functional-tests/tests/oauth/oauthPromptNone.spec.ts:67]: Does this line need a `// eslint-disable-line camelcase` too (re: `login_hint`)?

**Review by pdehaan — APPROVED** [review]
LGTM!
Comments were all just nits, so feel free to ignore if we want to land this ASAP (and we can tweak in the future).

**Comment by ashrivastava-qa** [comment]
Thanks @pdehaan I'll make the nit changes and merge it.

**Pull request merged**